src/benchq/algorithms/ld_gsee.py

```diff
@@ -0,0 +1,100 @@
+    arXiv:2209.06811v2).
+
+    See get_ldgsee_num_iterations for argument descriptions.
+
+    Returns: The standard deviation of the Gaussian convolution function."""
+    delta = precision**alpha * energy_gap ** (1 - alpha)
+    return min(
+        0.9
+        * delta
+        / (np.sqrt(2 * np.log(9 * delta * precision ** (-1) * square_overlap ** (-1)))),
+        0.2 * delta,
+    )
+
+
+def _get_epsilon_1(precision: float, square_overlap: float, sigma: float) -> float:
+    """Get the error with respect to the convolution function (Eq. 28 of
+        arXiv:2209.06811v2).
+
+    See get_ldgsee_num_iterations for argument descriptions.
+
+    Returns: The error with respect to the convolution function."""
+
+    return 0.1 * precision * square_overlap / (np.sqrt(2 * np.pi) * sigma**3)
+
+
+def get_ld_gsee_max_evolution_time(
+    alpha: float, energy_gap: float, square_overlap: float, precision: float
+) -> float:
+    """Get the maximum evolution time for the FF-LD-GSEE algorithm (Eq. 34 of
+    arXiv:2209.06811v2).
+
+    Arguments:
+        alpha: The parameter alpha controlling the tradeoff between circuit repetitions
+            and circuit depth. Zero corresponds to the minimum circuit depth, while one
+            corresponds to the minimum number of circuit repetitions.
+        energy_gap: The energy gap of the Hamiltonian, corresponds to Delta_true in the
+            paper.
+        square_overlap: The square overlap of the initial state with the ground state.
+            Corresponds to eta in the paper.
+        precision: The desired ground state energy precision. Corresponds to epsilon in
+            the paper.
+
+    Returns: The estimated number of iterations required by the FF-LD-GSEE algorithm."""
+    sigma = _get_sigma(alpha, energy_gap, square_overlap, precision)
+    epsilon_1 = _get_epsilon_1(precision, square_overlap, sigma)
+    return (
+        np.pi ** (-1)
+        * sigma ** (-1)
+        * np.sqrt(2 * np.log(8 * np.pi ** (-1) * epsilon_1 ** (-1) * sigma ** (-2)))
+    )
+
+
+def get_ld_gsee_num_circuit_repetitions(
+    alpha: float,
+    energy_gap: float,
+    square_overlap: float,
+    precision: float,
+    failure_probability: float,
+) -> float:
+    """Get the number of circuit repetitions for the FF-LD-GSEE algorithm (Eqs. 52 and
+    62 of arXiv:2209.06811v2).
+
+    Arguments:
+        alpha: The parameter alpha controlling the tradeoff between circuit repetitions
+            and circuit depth. Zero corresponds to the minimum circuit depth, while one
+            corresponds to the minimum number of circuit repetitions.
+        energy_gap: The energy gap of the Hamiltonian, corresponds to Delta_true in the
+            paper.
+        square_overlap: The square overlap of the initial state with the ground state.
+            Corresponds to eta in the paper.
+        precision: The desired ground state energy precision. Corresponds to epsilon in
```

Comment:
Let's add the units, which are Hartrees

Reply:
I updated the docstrings in c0d9f7925be3f69afec212c8cd7b6445ee5b4f92 to say that energy_gap and precision should have the same units. I don't think we need to specify Hartrees though since there is no assumption about what units actually are used.